### ISSUE TITLE
Trying to configure `licensecheck.licenses` property results in "Validation failed. 'value' length (.....) is longer than the maximum authorized (4000)"

### ISSUE THREAD

**Issue by oliverlockwood** [porscheinformatik/sonarqube-licensecheck#222]
See screenshot, where I've tried to add a few new characters into the existing property:
![Screenshot 2021-03-02 at 09 55 05](https://user-images.githubusercontent.com/6017680/109635303-1e385b80-7b42-11eb-9e2b-4504862f3e1c.png)

As the screenshot shows, the property is already configured to be a larger value (presumably on plugin install?), but the UI isn't letting me change it.  

We have recently upgraded from an old Sonar version (6.7.7) to 8.5.1.  The consensus in a similar-sounding issue https://github.com/SonarOpenCommunity/sonar-cxx/issues/1548 is that it's a new restriction introduced by Sonar, likely in https://github.com/SonarSource/sonarqube/commit/e9de4d178493dc10e411e9135159319e8e6873cb#diff-00b336cdea1cc5c1c7d29cb5578550e8.

However the fact remains it affects users of this plugin.

For now I'm going to look into manually editing the property in the DB (as suggested on the issue linked above) but that's not very maintainable, and I can imagine Sonar subsequently introducing harder limits that aren't only in the UI.  So it seems like this project either needs to:
- find a way to define the field in Sonar to allow it to hold more than 4000 characters (may not be possible?), or
- restructure configuration in such a way that the additional data can be input in another way, e.g.:
   - allowing `1...n` configurations which are appended to each other?
   - a completely different configuration model?

I look forward to your thoughts on this.  Thanks.

**Comment by oliverlockwood** [comment]
Workaround is as follows:
- get onto the Sonar DB (in our case, postgres)
- Running `SELECT * FROM properties WHERE prop_key = 'licensecheck.licenses';` should give you the current value just fine

**Comment by ricardojdsilva87** [comment]
Hello @oliverlockwood,
I'm not sure how you are deploying the sonarqube instance. But if it is using an helm chart you can load all of these licenses using a YAML file and setting the values on the sonar.properties file.
Currently we are setting them like this on the helm chart, where the env variables are the JSON files:
```
--set sonarProperties."licensecheck\.licenses"="${LICENSES}" \
--set sonarProperties."licensecheck\.licensesregex"="${LICENSES_REGEX}" \
--set sonarProperties."licensecheck\.alloweddependencies"="${ALLOWED_DEPENDENCIES}" \
```

If you go to the settings on sonarqube you can see that these can be mapped properties, like you can do in most of the settings.
![image](https://user-images.githubusercontent.com/12953702/111601898-baad5f80-87ca-11eb-9f76-d0a15225f8d7.png)

I agree with you, this should be configurable also via UI.
Thanks

**Comment by oliverlockwood** [comment]
Thanks @ricardojdsilva87, we do indeed use Helm.  I'm looking into your proposed approach.

**Comment by ricardojdsilva87** [comment]
Hello @oliverlockwood you should upgrade to the latest v5.0.0.
It changes the way that the data is handled, for now it converts from the old to the new data format so you have on the configuration something like this instead of a json file:
![image](https://user-images.githubusercontent.com/12953702/149902438-0c0b3e68-f6ac-4b3e-81f0-07adf4813056.png)
I did some tests and the UI is working nicely to add new licenses and dependencies and also after a reboot the changes are persistent, no need to handle the configurations on separate files

**Comment by Janpopan** [comment]
Has anybody try to use the sonar plugin api for adding licenses or project licenses. If I Try to do this I have to add all licenses or the actuall setting will replaced

**Comment by Janpopan** [comment]
@derkoe is the root cause behind this issue fixed with version 5.0.0?

**Comment by derkoe** [comment]
Yes this is fixed with 5.0.0 - thx for the hint.
